File: src/main/java/jokatu/game/joining/PlayerJoinedEvent.java
package jokatu.game.joining;

import jokatu.game.event.PublicGameEvent;
import jokatu.game.player.Player;
import org.jetbrains.annotations.NotNull;

/**
 * @author Steven Weston
 */
public class PlayerJoinedEvent implements PublicGameEvent {

	private final String message;

	public PlayerJoinedEvent(Player player) {
		message = player.getName() + " joined the game.";
	}

	@NotNull
	@Override
	public String getMessage() {
		return message;
	}
}

File: Core/src/main/kotlin/jokatu/game/joining/PlayerJoinedEvent.kt
package jokatu.game.joining

import jokatu.game.event.PublicGameEvent
import jokatu.game.player.Player

/**
 * An event for when a player joins a game.
 * @author Steven Weston
 */
class PlayerJoinedEvent(player: Player) : PublicGameEvent {
	override val message = "${player.name} joined the game."
}
